nginx.master.conf

```diff
@@ -156,21 +156,29 @@ http {
             proxy_pass http://navstar/lashup/key;
         }
 
+        location ~ ^/(slave|agent)/(?<agentid>[0-9a-zA-Z-]+)$ {
+            # Append slash and perform internal-redirect.
+            rewrite ^/(slave|agent)/(.*)$ /agent/$2/ last;
+        }
+
         location ~ ^/(slave|agent)/(?<agentid>[0-9a-zA-Z-]+)(?<url>.*)$ {
             access_by_lua 'auth.validate_jwt_or_exit()';
+
             set $agentaddr '';
             set $agentport '';
-
-            more_clear_input_headers Accept-Encoding;
             rewrite ^/(slave|agent)/[0-9a-zA-Z-]+/.*$ $url break;
             rewrite_by_lua_file conf/master/agent.lua;
 
+            more_clear_input_headers Accept-Encoding;
             include common/proxy-headers.conf;
-            include common/server-sent-events.conf;
-
-            # Supports chunked encoding and stream for debugging API
-            proxy_request_buffering off;
-
+            # The HTTP/1.1 does not matter as we do not use backend connection
+            # pooling anyway, and we need it enabled for streaming stdin/stdout
+            # to/from tasks.
+            include common/http-11.conf;
+            # We are OK with disabling all buffering here, as Mesos Agent does
+            # not care about it for normal endpoints while at the same time
```

Comment:
Well, this still does not give me the understanding that I feel I need to answer with "yes", but that's OK let's move ahead with this :-).